src/globus_cli/login_manager/tokenstore.py

```diff
@@ -10,15 +11,218 @@
 from .scopes import CURRENT_SCOPE_CONTRACT_VERSION
 
 if t.TYPE_CHECKING:
-    from globus_sdk.tokenstorage import SQLiteAdapter
+    from globus_sdk.tokenstorage import SQLiteAdapter  # noqa: F401
 
 # env vars used throughout this module
 GLOBUS_ENV = os.environ.get("GLOBUS_SDK_ENVIRONMENT")
 
 
-# stub to allow type casting of a function to an object with an attribute
-class _TokenStoreFuncProto:
-    _instance: SQLiteAdapter
+class CLITokenstorage:
+    """
+    A wrapper over the globus-sdk's v1 tokenstorage which provides simplified
+    capabilities specific to the CLI.
+    """
+
+    def __init__(self) -> None:
+        self._adapter: SQLiteAdapter | None = self._construct_adapter()
+
+    def _construct_adapter(self) -> SQLiteAdapter:
+        from globus_sdk.tokenstorage import SQLiteAdapter  # noqa: F811
+
+        # when initializing the token storage adapter, check if the storage file exists
+        # if it does not, then use this as a flag to clean the old config
+        fname = _get_storage_filename()
+        if not os.path.exists(fname):
+            from ._old_config import invalidate_old_config
+
+            invalidate_old_config(self.internal_native_client)
+
+        return SQLiteAdapter(fname, namespace=_resolve_namespace())
+
+    def close(self) -> None:
+        if self._adapter is not None:
+            self._adapter.close()
+            self._adapter = None
+
+    @functools.cached_property
+    def internal_native_client(self) -> globus_sdk.NativeAppAuthClient:
+        """
+        This is the client that represents the CLI itself (prior to templating).
+        """
+        template_id = _template_client_id()
+        return globus_sdk.NativeAppAuthClient(
+            template_id, app_name="Globus CLI (native client)"
+        )
+
+    def internal_auth_client(self) -> globus_sdk.ConfidentialAppAuthClient:
```

Comment:
The name `internal_*` sounds like it's exposing the internals of the class. What do you think about using the term `common_*`? Or even just drop the prefix (`token_storage.native_client`)?

Additionally I'd recommend using `confidential` rather than `auth` in the second one to more accurately highlight the differences between these two. 


```suggestion
    def common_native_client(self) -> globus_sdk.NativeAppAuthClient:
        """
        This is the client that represents the CLI itself (prior to templating).
        """
        template_id = _template_client_id()
        return globus_sdk.NativeAppAuthClient(
            template_id, app_name="Globus CLI (native client)"
        )

    def common_confidential_client(self) -> globus_sdk.ConfidentialAppAuthClient:
```

Reply:
_Alternatively if these are not meant to be shared, I'd recommend `_native_client` and `_confidential_client`_

Reply:
We talked it through; going with `cli_{thing}`, and improving the confidential client docstring to give some better explanation.